src/graphnet/models/components/layers.py

```diff
@@ -59,3 +65,140 @@ def forward(
         ).to(self.device)
 
         return x, edge_index
+
+
+class EdgeConv0(MessagePassing, LightningModule):
+    """Implementation of EdgeConv0 layer used in TITO solution for.
+
+    'IceCube - Neutrinos in Deep' kaggle competition.
+    """
+
+    def __init__(
+        self,
+        nn: Callable,
+        aggr: str = "max",
+        **kwargs: Any,
+    ):
+        """Construct `EdgeConv0`.
+
+        Args:
+            nn: The MLP/torch.Module to be used within the `EdgeConv0`.
+            aggr: Aggregation method to be used with `EdgeConv0`.
+            **kwargs: Additional features to be passed to `EdgeConv0`.
+        """
+        super().__init__(aggr=aggr, **kwargs)
+        self.nn = nn
+        self.reset_parameters()
+
+    def reset_parameters(self) -> None:
+        """Reset all learnable parameters of the module."""
+        reset(self.nn)
+
+    def forward(self, x: Union[Tensor, PairTensor], edge_index: Adj) -> Tensor:
+        """Forward pass."""
+        if isinstance(x, Tensor):
+            x = (x, x)
+        # propagate_type: (x: PairTensor)
+        return self.propagate(edge_index, x=x, size=None)
+
+    def message(self, x_i: Tensor, x_j: Tensor) -> Tensor:
+        """EdgeConv0 message passing."""
+        return self.nn(torch.cat([x_i, x_j - x_i, x_j], dim=-1))  # edgeConv0
+
+    def __repr__(self) -> str:
+        """Print out module name."""
+        return f"{self.__class__.__name__}(nn={self.nn})"
+
+
+class DynTrans(EdgeConv0, LightningModule):
+    """Implementation of dynTrans1 layer used in TITO solution for.
+
+    'IceCube - Neutrinos in Deep' kaggle competition.
+    """
+
+    def __init__(
+        self,
+        layer_sizes: Optional[List[int]] = None,
+        aggr: str = "max",
+        nb_neighbors: int = 8,
```

Comment:
It appears that `nb_neighbors` is unused. Am I missing something?

Reply:
You are right, I'll remove it.